### PR TITLE
Compose functional Chainweb requests

### LINKED ISSUE #144: Compose functional Chainweb requests
ℹ️ This issue is part of Epic #57

## Objective
- Implement a module to perform requests to Chainweb.

## Checklist
- [x] Initialization function `new/2` receiving the **HTTP method** and the **type of Chainweb API** (`pact`, `p2p`)
- [x] Functions to set **chain id** and **network**
- [x] Functions to add **body** and **headers** to the request.
- [x] Function `perform/1` to make the HTTP request once configured.
- [x] `results/2` to represent the API response as a defined structure.

### PR DESCRIPTION
**Pull request by miguelnietoa**
## Description

Closes #144 

<details>
<summary>Example</summary>
<br>

```elixir
endpoint = "local"
network_id = :testnet04
chain_id = "0"
body = "{\"cmd\":\"{\\\"meta\\\":{\\\"chainId\\\":\\\"0\\\",\\\"creationTime\\\":1667249173,\\\"gasLimit\\\":1000,\\\"gasPrice\\\":1.0e-6,\\\"sender\\\":\\\"k:554754f48b16df24b552f6832dda090642ed9658559fef9f3ee1bb4637ea7c94\\\",\\\"ttl\\\":28800},\\\"networkId\\\":\\\"testnet04\\\",\\\"nonce\\\":\\\"2023-06-13 17:45:18.211131 UTC\\\",\\\"payload\\\":{\\\"exec\\\":{\\\"code\\\":\\\"(+ 5 6)\\\",\\\"data\\\":{}}},\\\"signers\\\":[{\\\"addr\\\":\\\"85bef77ea3570387cac57da34938f246c7460dc533a67823f065823e327b2afd\\\",\\\"clist\\\":[{\\\"args\\\":[\\\"85bef77ea3570387cac57da34938f246c7460dc533a67823f065823e327b2afd\\\"],\\\"name\\\":\\\"coin.GAS\\\"}],\\\"pubKey\\\":\\\"85bef77ea3570387cac57da34938f246c7460dc533a67823f065823e327b2afd\\\",\\\"scheme\\\":\\\"ED25519\\\"}]}\",\"hash\":\"-1npoTU2Mi71pKE_oteJiJuHuXTXxoObJm8zzVRK2pk\",\"sigs\":[{\"sig\":\"8b234b83570359e52188cceb301036a2a7b255171e856fd550cac687a946f18fbfc0e769fd8393dda44d6d04c31b531eaf35efb3b78b5e40fd857a743133030d\"}]}"
headers = [{"Content-Type", "application/json"}]

:post
  |> Request.new(pact: [endpoint: endpoint])
  |> Request.set_chain_id(chain_id)
  |> Request.set_network(network_id)
  |> Request.add_body(body)
  |> Request.add_headers(headers)
  |> Request.perform()
  |> Request.results(as: LocalResponse)
```

```elixir
{:ok,
 %Kadena.Chainweb.Pact.LocalResponse{
   continuation: nil,
   events: nil,
   gas: 7,
   logs: "wsATyGqckuIvlm89hhd2j4t6RMkCrcwJe_oeCYr7Th8",
   meta_data: %{
     block_height: 2815727,
     block_time: 1671054330981668,
     prev_block_hash: "asisNr3nuU0t357i2bxMVUiIWDVHAMtncJHtmyENbio",
     public_meta: %{
       chain_id: "0",
       creation_time: 1667249173,
       gas_limit: 1000,
       gas_price: 1.0e-6,
       sender: "k:554754f48b16df24b552f6832dda090642ed9658559fef9f3ee1bb4637ea7c94",
       ttl: 28800
     }
   },
   req_key: "-1npoTU2Mi71pKE_oteJiJuHuXTXxoObJm8zzVRK2pk",
   result: %{data: 11, status: "success"},
   tx_id: nil
 }}
```

</details>

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/chainweb/request.ex [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      path: path,
+      segment: segment,
+      segment_path: segment_path,
+      query: [],
+      headers: []
+    }
+  end
+
+  @spec set_chain_id(t(), chain_id :: chain_id()) :: t()
+  def set_chain_id(%__MODULE__{} = request, chain_id) do
```
Comment (L-Zuluaga) [inline lib/chainweb/request.ex]: Maybe is better to use the `, do: ` in these group of functions?

File: lib/chainweb/request.ex [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        } = request
+      ) do
+    request
+    |> build_request_url()
+    |> (&Client.request(method, &1, headers, body)).()
+  end
+
+  @spec results(response :: response(), opts :: opts()) :: parsed_response()
+  def results({:ok, results}, as: resource) do
+    {:ok, results |> MapCase.to_snake!() |> resource.new()}
```
Comment (L-Zuluaga) [inline lib/chainweb/request.ex]: 🤔 What do you think about changing it for something like this? 
```suggestion
    resource = 
         results 
         |> MapCase.to_snake!() 
         |> resource.new()
         
    {:ok, resource}
```

**Review by L-Zuluaga — COMMENTED** [review]
LGTM, I left some comments in case you want to take them into account 🚀

**Comment by miguelnietoa** [comment]
@L-Zuluaga Of course, done!

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**